include/jemalloc/internal/hpa.h

```diff
@@ -0,0 +1,70 @@
+#ifndef JEMALLOC_INTERNAL_HPA_H
+#define JEMALLOC_INTERNAL_HPA_H
+
+#include "jemalloc/internal/geom_grow.h"
+#include "jemalloc/internal/hpa_central.h"
+#include "jemalloc/internal/pai.h"
+#include "jemalloc/internal/psset.h"
+
+typedef struct hpa_s hpa_t;
+struct hpa_s {
+	malloc_mutex_t grow_mtx;
+	malloc_mutex_t mtx;
+	hpa_central_t central;
+	/* The arena ind we're associated with. */
+	unsigned ind;
+	edata_cache_t *edata_cache;
+	geom_grow_t geom_grow;
```

Comment:
We could, although I don't know how useful it'd be (maybe as a stat? "Ah, you're not getting any hugepages because all your arenas use custom extent hooks"?). Up to you.

Reply:
No need to do it now. Let's wait and see.